pm_icecon/bt/compute_bt_ic.py

```diff
@@ -724,24 +716,32 @@ def coastal_fix(
     is_considered = is_temp0 & is_rolled_land
 
     # args to np.roll are opposite of fortran index offsets
+    # TODO: one less roll operation here than in `_conc_change124`. Should there
+    # be?
     tip1jp1 = np.roll(temp, (-1, -1), axis=(1, 0))
     tip1jp0 = np.roll(temp, (-1, 0), axis=(1, 0))
     tip0jm1 = np.roll(temp, (0, 1), axis=(1, 0))
     tip0jp1 = np.roll(temp, (0, -1), axis=(1, 0))
 
+    # TODO: one less conditional here than in `_conc_change124`. Should there
+    # be?
     is_tip1jp1_le0 = tip1jp1 <= 0
     is_tip1jp0_eq0 = tip1jp0 == 0
     is_tip0jm1_le0 = tip0jm1 <= 0
     is_tip0jp1_le0 = tip0jp1 <= 0
 
     # Changing conc2(i+1,j) to 0
+    # TODO: `is_tip1jp1_le0` variable is used twice here (&-ed
+    # together). Was this a mistake?
```

Comment:
It seems like this might be a mistake. The double `&`-ed `is_tip1jp1_le0` here and below may be a stand-in for the missing conditional?

Reply:
This code is so convoluted that -- as you're seeing -- it's tricky to even understand what it's doing, let alone put it in a vector-math form.

Also -- if I understand what it is trying to do -- it is not efficiently programmed in a numerical sense.

I'll look this over in more detail tomorrow and early next week.  With this much nearly-repeating code, I suspect you have found a bug/typo.  I know there was at least one other that I found in the original bootstrap fortran a while back.